File: ruby-neo4j-driver/common_friends.rb
require './neo4j_init'

# Get session
session = Neo4jInit.getSession("../config.yml")

# Common friends
common_friends_query = <<QUERY
  MATCH (user:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf:Person)
  WHERE user.name = {user} AND foaf.name = {foaf}
  RETURN friend.name AS friend
QUERY

response = session.query(common_friends_query, user: 'Joe', foaf: 'Sally')
puts response.map(&:friend)
puts '---------------------'

File: elixir-neo4j-driver/social/common_friends.exs
[[{'userName', username}, {'password', password}, {'host', host}, _ , _, _]] = :yamerl_constr.file("../../config.yml")
{:ok, _neo} = Bolt.Sips.start_link(url: "bolt://#{username}:#{password}@#{host}")
conn = Bolt.Sips.conn()

# common friends
params = %{user: "Joe", foaf: "Sally"}
common_friends_query = """
  MATCH (user:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf:Person)
  WHERE user.name = {user} AND foaf.name = {foaf}
  RETURN friend.name AS friend
"""

Bolt.Sips.query!(conn, common_friends_query, params) |> IO.inspect
